### ISSUE TITLE
Ensure doc builds are selfcontained, not relying on astropy-dev

### ISSUE THREAD

**Issue by mhvk** [astropy/astropy#16531]
### What is the problem this feature will solve?

Currently, if the docs updates cause some references to disappear, the readthedocs build does not fail, as it gets references from intersphinx mappings: https://github.com/astropy/astropy/blob/fe7b649b433c057f1fa4ca638b60e0b7097a024d/docs/conf.py#L112-L114

This now has meant several times after a PR is merged, `main` will fail and there has to be a follow-up PR (most recently, #16513, followed up by #16530).

Can we avoid this? Is that line in `conf.py` for astropy-dev in fact needed?

### Describe the desired outcome

Tests ensure that when a PR is merger, main continues to pass CI.

**Comment by pllim** [comment]
We added this on purpose for a reason. I'll dig tomorrow.

**Comment by pllim** [comment]
So, `astropy-dev` was implemented back inhttps://github.com/astropy/astropy/pull/11622 to make devdocs cleaner.

Still unclear to me why this would cause problem for things not in devdocs though.

**Comment by mhvk** [comment]
Thanks for chasing that down, @pllim! So I think the essential element is,
> but particularly this is useful for linking to the Developer Guide since it now lives only in the
development version.

So, the goal is to ensure the link to the developer guide can be found also in the regular documentation.

This seems reasonable, but the problem is that somehow for missing links those docs are searched as well.  Maybe that is simply because those `astropy-dev` docs provide `~astropy`? Not quite sure, nor how to solve this...

**Comment by pllim** [comment]
> Maybe that is simply because those `astropy-dev` docs provide `~astropy`? 

I *think* (at least it works in my head), one workaround is for the PR author to temporarily re-direct `astropy-dev` URL to their own PR branch. And then once everything is done and we can make sure the final CI run is green, we revert that change. This does put the burden on PR authors and we also have to remember to squash-merge or have PR author clean up the commits.

**Comment by mhvk** [comment]
Ideally, we just make the work-around work only on stable, for which it is actually needed, because it no longer has the development documents. One option would be to just create a hard link to the development documents and rely on link checker to warn us if that ever goes out of date.

**Comment by pllim** [comment]
> create a hard link to the development documents

Are you proposing that we revert https://github.com/astropy/astropy/pull/11622 ?

**Comment by mhvk** [comment]
Maybe. I think hoping that contributors remember to change links and then change them back is unlikely to be less problematic than what we have now... Also, conceptually, cross-linking between different astropy versions does not feel particularly logical or easy to explain... But obviously any change has to be worth the effort.

A `git grep astropy-dev` finds four places where it is used to point to development docs:
```
docs/index_dev.rst:<astropy-dev:developer-docs>`.
docs/install.rst:into a :ref:`virtualenv <astropy-dev:virtual_envs>`.
docs/install.rst:astropy <astropy-dev:testing_installed_astropy>`.
docs/install.rst:checkout of astropy <astropy-dev:sourcebuildtest>`
```
The first is the old link that was changed (goes to https://docs.astropy.org/en/latest/index_dev.html).  I'm not sure any of the others belong in the installation part of the "getting started" section...

**Comment by pllim** [comment]
I don't remember if downstream uses `astropy-dev for intersphinx or not. Will have to check when I have time.

**Comment by pllim** [comment]
A quick search indicates that removing `astropy-dev` is probably okay as far as downstream is concerned. We do have to modify the docs here a little though, e.g., `docs/index_dev.rst`, `docs/conf.py`, `docs/development/docguide.rst`, etc.

**Comment by astrofrog** [comment]
Of course there's always the option of splitting out the dev docs into their own repo as a lot of them also apply to coordinated packages etc (this is an idea that has come up several times in the past)

**Comment by pllim** [comment]
Right, we considered splitting it out but decided not to because then we risk the doc out of sync with code.
